src/assessments/seo/textCompetingLinksAssessment.js

```diff
@@ -1,73 +1,116 @@
-var AssessmentResult = require( "../../values/AssessmentResult.js" );
+const Assessment = require( "../../assessment.js" );
+const AssessmentResult = require( "../../values/AssessmentResult.js" );
 
-var Mark = require( "../../values/Mark.js" );
-var addMark = require( "../../markers/addMark.js" );
+const Mark = require( "../../values/Mark.js" );
+const addMark = require( "../../markers/addMark.js" );
 
-var map = require( "lodash/map" );
+const map = require( "lodash/map" );
+const merge = require( "lodash/merge" );
+const isUndefined = require( "lodash/isUndefined" );
 
 /**
- * Returns a score and text based on the number of links.
- *
- * @param {object} linkStatistics The object with all linkstatistics.
- * @param {object} i18n The object used for translations
- * @returns {object} resultObject with score and text
+ * Assessment to check whether the keyphrase is encountered in the first paragraph of the article.
  */
-var calculateLinkCountResult = function( linkStatistics, i18n ) {
-	if ( linkStatistics.keyword.totalKeyword > 0 ) {
-		return {
-			score: 2,
-			hasMarks: true,
-			text: i18n.dgettext( "js-text-analysis", "You're linking to another page with the focus keyword you want this page to rank for. " +
-				"Consider changing that if you truly want this page to rank." ),
+class TextCompetingLinksAssessment extends Assessment {
+	/**
+	 * Sets the identifier and the config.
+	 *
+	 * @param {Object} config The configuration to use.
+	 *
+	 * @returns {void}
+	 */
+	constructor( config = {} ) {
+		super();
+
+		let defaultConfig = {
+			parameters: {
+				recommendedMaximum: 0,
+			},
+			bad: {
+				score: 2,
+				resultText: "You're linking to another page with the focus keyword you want this page to rank for. " +
+				"Consider changing that if you truly want this page to rank.",
+			},
 		};
+
+		this.identifier = "textCompetingLinks";
+		this._config = merge( defaultConfig, config );
 	}
-	return {};
-};
 
-/**
- * Runs the linkCount module, based on this returns an assessment result with score.
- *
- * @param {object} paper The paper to use for the assessment.
- * @param {object} researcher The researcher used for calling research.
- * @param {object} i18n The object used for translations
- * @returns {object} the Assessmentresult
- */
-var textHasCompetingLinksAssessment = function( paper, researcher, i18n ) {
-	var linkCount = researcher.getResearch( "getLinkStatistics" );
+	/**
+	 * Runs the linkCount module, based on this returns an assessment result with score.
+	 *
+	 * @param {object} paper The paper to use for the assessment.
+	 * @param {object} researcher The researcher used for calling research.
+	 * @param {object} i18n The object used for translations
```

Comment:
I'd put a blank line between the parameters and `@returns`.  Also capitalize `{Object}`.

Reply:
Fixed.